recipes-core/date-sync/init

```diff
@@ -1,42 +1,76 @@
 #!/bin/sh
-#
 ### BEGIN INIT INFO
-# Provides:		date-sync
-# Required-Start:	$remote_fs $syslog $networking
-# Required-Stop:	$remote_fs $syslog
-# Default-Start:	2 3 4 5
-# Default-Stop:		1
-# Short-Description:	Start and stop the date-sync daemon
+# Provides:             date-sync
+# Required-Start:       $remote_fs $syslog $networking
+# Required-Stop:        $remote_fs $syslog
+# Default-Start:        2 3 4 5
+# Default-Stop:         1
+# Short-Description:    Start and stop the date-sync daemon
 ### END INIT INFO
-#
 
 PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
 NAME=date-sync
 DESC="Date Sync"
 LOGFILE=/tmp/date-sync.log
+PIDFILE=/var/run/date-sync.pid
 
-start() {
-	echo -n "Starting $DESC: "
+# Configuration
+NTP_SERVER="pool.ntp.org"
+SYNC_INTERVAL=60  # Run every 60 seconds (1 minute)
+MAX_LOG_SIZE=524288  # 512KB (0.5MB) in bytes
 
-	# update time by fetching the latest one from ntp
-	ntpd -q -p pool.ntp.org
+log() {
+    echo "$(date +"%Y-%m-%d %H:%M:%S"): $1" >> $LOGFILE
+    if [ -f "$LOGFILE" ] && [ $(stat -c %s "$LOGFILE") -gt $MAX_LOG_SIZE ]; then
+        tail -n 1000 "$LOGFILE" > "${LOGFILE}.tmp" && mv "${LOGFILE}.tmp" "$LOGFILE"
+    fi
+}
 
-	# get current time
-	current_time=$(date)
+sync_time() {
+    log "Synchronizing time with $NTP_SERVER"
+    if ntpd -q -p $NTP_SERVER; then
+        log "Time synchronized successfully"
```

Comment:
You could make those into a single log for less bloat --
`log "Time synchronized with $NTP_SERVER successfully"`
same for failure.

You also don't really need to lo $NTP_SERVER since it's constant
What would be really nice to log is the drift. If you want to do this I'd probably check out:
1. `-l logfile` option of ntpd - to move its logs to $LOGFILE
2. `-D level` option for increasing verbosity in case it doesn't log the drift by detault, maybe it does on higher verbosity

Reply:
Those flags are not available in busybox ntpd but I've simulated them as much as possible.
Good idea to merge the logs.